pyccel/codegen/printing/ccode.py

```diff
@@ -953,8 +954,11 @@ def _print_FunctionDef(self, expr):
             self._additional_args.append(expr.results)
         body  = self._print(expr.body)
         decs  = [Declare(i.dtype, i) if isinstance(i, Variable) else FuncAddressDeclare(i) for i in expr.local_vars]
-        if len(expr.results) <= 1 :
-            decs += [Declare(i.dtype, i) if isinstance(i, Variable) else FuncAddressDeclare(i) for i in expr.results]
+        if len(expr.results) == 1 :
+            if isinstance(expr.results[0], Variable) and expr.results[0] != self._unused_variable:
+                decs += [Declare(expr.results[0].dtype, expr.results[0])]
+            elif isinstance(expr.results[0], FunctionAddress):
+                decs += [FuncAddressDeclare(expr.results[0])]
```

Comment:
If `_unused_variable` is to be reusable for anything other than return variables (its name does not indicate that it is only useful for returns so I assume it is general) then the big if here should be done for every declaration. Including the declarations in the first line of this code

Reply:
its a great i idea to have something that holds unused variables and it will even better if we have the ability to detect unused variables especially in local variables, but this variable in particular is specific in holding the unused variable that is generated from the return statement, So i think i will change its name to be more clear, and for holding unused variables in general i think is to be implemented.

Reply:
If you only plan for this object to hold unused variables generated by the return statement then renaming it is important in order for that to be clear.